app/models/dataset_record/darwin_core/occurrence.rb

```diff
@@ -827,37 +827,80 @@ def parse_location_class
     }
   end
 
-  def parse_identification_class
-    taxon_determination = {}
+  def parse_typestatus(type_status)
     type_material = nil
+    type_status_parsed = type_status&.match(/^(?<type>\w+)$/i) || type_status&.match(/(?<type>\w+)(\s+OF\s+(?<scientificName>.*))/i)
 
-    # identificationID: [Not mapped]
+    # only nil if non-alphanumeric entry, or multiple words not matching "\w+ of \w+"
+    raise DarwinCore::InvalidData.new({ "typeStatus": ["Unprocessable typeStatus information"] }) unless type_status_parsed
 
-    # identificationQualifier: [Mapped as part of otu name in parse_taxon_class]
+    code = get_field_value(:nomenclaturalCode)&.downcase&.to_sym || import_dataset.default_nomenclatural_code
+    unless TypeMaterial::legal_type_type(code, type_status_parsed[:type].downcase)
+      raise DarwinCore::InvalidData.new({ "typeStatus": ["could not extract legal type from typeStatus"] })
+    end
 
-    # typeStatus: [Type material only if scientific name matches scientificName and type term is recognized by TW vocabulary]
-    type_status = get_field_value(:typeStatus)
-    type_status_parsed = type_status&.match(/^(?<type>\w+)$/i) || type_status&.match(/(?<type>\w+)(\s+OF\s+(?<scientificName>.*))/i)
     scientific_name = get_field_value(:scientificName)&.gsub(/\s+/, ' ')
-    type_scientific_name = (type_status_parsed&.[](:scientificName)&.gsub(/\s+/, ' ') rescue nil) || scientific_name
 
-    if type_status_parsed && scientific_name && type_scientific_name.present?
+    # if typeStatus is single word, assume the user wants the specimen name as the type name
+    type_scientific_name = (type_status_parsed&.[](:scientificName)&.gsub(/\s+/, ' ') rescue nil) || scientific_name
 
+    if scientific_name && type_scientific_name.present?
       # if type_scientific_name matches the current name of the occurrence, use that
       if type_scientific_name.delete_prefix(scientific_name)&.match(/^\W*$/)
-        type_material = {
+        return {
           type_type: type_status_parsed[:type].downcase
         }
-      elsif (original_combination_protonym = Protonym.find_by(cached_original_combination: type_scientific_name, project_id: self.project_id))
-        type_material = {
+      end
+      if (original_combination_protonym = Protonym.find_by(cached_original_combination: type_scientific_name, project_id: self.project_id))
+        return {
           type_type: type_status_parsed[:type].downcase,
           protonym: original_combination_protonym
         }
       end
+
+      # Try wildcard match on subgenus if not present
+      type_name_elements = type_scientific_name.split
+      if type_name_elements.length > 1 && type_name_elements[1].first != "(" && type_name_elements[1].last != ")"
+
+        # append subgenus wildcard to genus string
```

Comment:
Just for peace of mind can you use `Regexp.escape` on each name element before proceeding with next sentences up to `Protonym.where`?